test/api/routing.test.js

```diff
@@ -81,4 +87,44 @@ describe('Routing', () => {
       });
     });
   });
-});
+
+  context('getMatrixDistances', () => {
+    describe('location permissions denied', () => {
+      it('should throw a navigation error', () => {
+        navigatorStub.rejects(STATUS.ERROR_PERMISSIONS);
+
+        return Matrix.getMatrixDistances()
+          .catch((err) => {
+            expect(err.toString()).to.eq(STATUS.ERROR_PERMISSIONS);
+            expect(httpStub).to.have.callCount(0);
+          });
+      });
+    });
+
+    describe('location permissions approved', () => {
+      describe('no args given', () => {
+        it('should return a routing response', () => {
+          navigatorStub.resolves(matrixOrigin);
+          httpStub.resolves(matrixResponse);
+
+          return Matrix.getMatrixDistances()
+            .then((response) => {
+              expect(response).to.equal(matrixResponse);
+            });
+        });
+      });
+
+      describe('all args given', () => {
+        it('should return a routing response', () => {
+          navigatorStub.resolves(matrixOrigin);
+          httpStub.resolves(matrixResponse);
+
+          return Matrix.getMatrixDistances({ matrixDestination, matrixMode, units })
```

Comment:
ah these don't look like the right input names

`{ matrixDestination, matrixMode, units }`

the actual names are `{ origins,  destinations, mode, units }`
can do `getMatrixDistances({ destinations: MatrixDestination, mode: matrixMode })` if the variable name doesn't align with the input